pkg/kubelet/userns/inuserns: use moby/sys/userns

kolyshkin opened the pull request:
The code from github.com/opencontainers/runc/libcontainer/userns package was moved into github.com/moby/sys/user and github.com/moby/sys/userns (see [1]), and the runc package is now deprecated in favor of moby/sys (see [2]).

Let's switch to moby/sys/userns.

[1]: https://github.com/moby/sys/releases/tag/userns%2Fv0.1.0
[2]: https://github.com/opencontainers/runc/pull/4350

#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

kolyshkin committed:
pkg/kubelet/user/userns: remove, use moby/sys/userns

The code from github.com/opencontainers/runc/libcontainer/userns package
was moved into github.com/moby/sys/user and github.com/moby/sys/userns
(see [1]), and the runc package is now deprecated in favor of moby/sys
(see [2]).

In addition, moby/sys/userns now has a non-Linux implementation, so
pkg/kubelet/user/userns package (introduced in commit 2e999ff to make a
non-Linux implementation) is not really needed anymore.

Let's switch to moby/sys/userns, and remove the package.

[1]: https://github.com/moby/sys/releases/tag/userns%2Fv0.1.0
[2]: https://github.com/opencontainers/runc/pull/4350

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

AkihiroSuda commented on pkg/kubelet/userns/inuserns/inuserns_linux.go:
```diff
@@ -16,9 +16,9 @@ limitations under the License.
 
 package inuserns
```
This pkg can be probably removed

kolyshkin replied:
I thought about it, but we don't know about external dependencies.

Deprecated it.

kolyshkin replied:
Ah, I see this was only added recently (commit 2e999fff026) so makes sense to remove entirely.

dims commented:
/release-note-none

dims commented:
@kolyshkin looks like one folder wasn't added to the PR

```
Generated files need to be updated
Not currently on any branch.
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	LICENSES/vendor/github.com/moby/sys/userns/

nothing added to commit but untracked files present (use "git add" to track)

Please run 'hack/update-vendor-licenses.sh'

```

dims commented:
/release-note-none

dims commented:
As expected adding one dependency

```
-Direct Dependencies: 188 
+Direct Dependencies: 189 
 Transitive Dependencies: 333 
-Total Dependencies: 371 
+Total Dependencies: 372 
 Max Depth Of Dependencies: 25 
 All dependencies:
 
@@ -256,6 +256,7 @@
 github.com/moby/ipvs
 github.com/moby/spdystream
 github.com/moby/sys/mountinfo
+github.com/moby/sys/userns
 github.com/moby/term
 github.com/modern-go/concurrent
 github.com/modern-go/reflect2
```

dims commented:
cc @derekwaynecarr @mrunalp @SergeyKanzhelev

kolyshkin committed:
staging/src/k8s.io/mount-utils: use moby/sys/userns

This package was moved out from runc/libcontainer and it will be made
deprecated.

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

dims commented:
/approve
/lgtm

straight forward replace
